### ISSUE TITLE
SyntaxError: Unexpected end of JSON input returns statuscode 500 instead of 400

### ISSUE THREAD

**Issue by lo78cn** [nestjs/nest#430]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
`curl -X POST "http://localhost:3000/login" -H "accept: application/json" -H "Content-Type: application/json" -d "{ \"email\": \"user@domain.com\", \"password\": \"secret\""`
```
{
  "statusCode": 500,
  "message": "Internal server error"
}
```

## Expected behavior
`curl -X POST "http://localhost:3000/login" -H "accept: application/json" -H "Content-Type: application/json" -d "{ \"email\": \"user@domain.com\", \"password\": \"secret\""`
```
{
  "statusCode": 400,
  "message": "Bad request"
}
```
Or even better:
```
{
  "statusCode": 400,
  "message": "SyntaxError: Unexpected end of JSON input"
}
```
## Minimal reproduction of the problem with instructions
Boilerplate nestjs application, any post endpoint, any invalid json POST. 

## What is the motivation / use case for changing the behavior?
If I enable my global AnyExceptionFilter and log the exception, I get a 400 statuscode.
```
{ SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at parse (/Users/name/Downloads/avgAPI/node_modules/body-parser/lib/types/json.js:88:17)
    at /Users/name/Downloads/avgAPI/node_modules/body-parser/lib/read.js:116:18
    at invokeCallback (/Users/name/Downloads/avgAPI/node_modules/raw-body/index.js:262:16)
    at done (/Users/name/Downloads/avgAPI/node_modules/raw-body/index.js:251:7)
    at IncomingMessage.onEnd (/Users/name/Downloads/avgAPI/node_modules/raw-body/index.js:307:7)
    at emitNone (events.js:106:13)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1056:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
  body: '{\n  "email": "user@domain.com",\n  "password": "secret"',
  status: 400,
  statusCode: 400 }
```

## Environment

<pre><code>
Nest version: 4.6.3

 
For Tooling issues:
- Node version: 8.9.1  
- Platform:  Mac 

Others:

</code></pre>

**Comment by chanlito** [comment]
@lo78cn use exception filter.

**Comment by lo78cn** [comment]
@chanlito I disagree.
```
Internal Error 500
The server encountered an unexpected condition which prevented it from fulfilling the request.
```
It is not a unexpected.
```
Bad request 400
The request had bad syntax or was inherently impossible to be satisfied.
``` 
Makes more sense in my opinion.

**Comment by kamilmysliwiec** [comment]
Fixed in `v4.6.4` 🙂
